File: client/src/dto/PlayerInfoBundleDto.ts
import { PartialPlayerGameInfoDto, undefinedPlayerGameInfo } from "./PartialPlayerGameInfoDto";

// Used to populate the match's player info context
export interface PlayerGameInfoBundleDto {
    player1GameInfo: PartialPlayerGameInfoDto;
    player2GameInfo: PartialPlayerGameInfoDto;
}

export const undefinedPlayerInfoBundleDto: PlayerGameInfoBundleDto = {
    player1GameInfo: undefinedPlayerGameInfo,
    player2GameInfo: undefinedPlayerGameInfo
};

File: server/dto/player_info_bundle_dto.py
from dataclasses import dataclass

from dto.base_dto import BaseDto
from dto.partial_player_game_info_dto import PartialPlayerGameInfoDto


@dataclass
class PlayerGameInfoBundleDto(BaseDto):
    player1GameInfo: PartialPlayerGameInfoDto
    player2GameInfo: PartialPlayerGameInfoDto
